python/lsst/analysis/tools/actions/plot/propertyMapPlot.py

```diff
@@ -506,112 +614,232 @@ def getOutputNames(self, config=None) -> Iterable[str]:
+            ha="left",
+            va="top",
+        )
+        lineHeightFraction = title.get_fontsize() / (fig.get_size_inches()[1] * fig.dpi)
+        infoBoxTopLeftCorner = (titleBoxTopLeftCorner[0], titleBoxTopLeftCorner[1] - 1.8 * lineHeightFraction)
+        info = fig.text(
+            *infoBoxTopLeftCorner,
+            f"{run}{tableType}{geomText}{infoText}",
+            fontsize=15,
+            transform=fig.transFigure,
+            alpha=0.6,
+            ha="left",
+            va="top",
         )
+        info.set_linespacing(1.4)
 
-    @staticmethod
-    def getLongestSuffixMatch(s, options):
-        """Find the longest suffix in the provided list that matches the end of
-        the given string.
+        return fig
+
+    def makePlot(
+        self,
+        data: KeyedData,
+        plotConfig: PropertyMapSurveyWideAnalysisConfig,
+        plotInfo: Mapping[str, Union[Mapping[str, str], str, int]],
+    ) -> Figure:
+        """Make the survey property map plot.
 
         Parameters
         ----------
-        s : `str`
-            The target string for which we want to find a matching suffix.
-        options : `list` [`str`]
-            A list of potential suffix strings to match against the target
-            string `s`.
+        data : `KeyedData`
+            The HealSparseMap to plot the points from.
+        plotConfig :
+            `~lsst.analysis.tools.tasks.propertyMapSurveyAnalysis.
+            PropertyMapSurveyWideAnalysisConfig`
+            The configuration for the plot.
+        plotInfo : `dict`
+            A dictionary of information about the data being plotted.
 
         Returns
         -------
-        `str`
-            The longest matching suffix from the `options` list. If no match is
-            found, returns `None`.
+        fig : `~matplotlib.figure.Figure`
+            The resulting figure.
         """
-        return next((opt for opt in sorted(options, key=len, reverse=True) if s.endswith(opt)), None)
+
+        # 'plotName' defaults to the attribute specified in
+        # 'atools.<attribute>' in the pipeline YAML. If it is explicitly
+        # set in `~lsst.analysis.tools.atools.healSparsePropertyMap.
+        # HealSparsePropertyMapTool`, it will override this default.
+        if self.plotName:
+            # Set the plot name using 'produce.plot.plotName' from
+            # HealSparsePropertyMapTool's instance.
+            plotInfo["plotName"] = self.plotName
+
+        # Plotting customization.
+        colorbarTickLabelSize = 16
+        rcparams = {
+            "axes.labelsize": 18,
+            "axes.linewidth": 1.8,
+            "xtick.labelsize": 13,
+            "ytick.labelsize": 13,
+        }
+
+        mapName = data["data"].ref.datasetType.name
+        mapData = data["data"].get()
+
+        with plt.rc_context(rcparams):
+            # The figsize should be decided based on the survey footprint.
+            fig, ax = plt.subplots(1, 1, figsize=(19, 7))
```

Comment:
Can you try removing the plt? The code is pretty slow and it would be useful to speed it up. The property map code in general is slow.

Reply:
Is the `with plt.rc_context(rcparams)` part causing the slow-down? If I remember right, this is how @erykoff  suggested setting `rcparams`. I mentioned him to see if he has any thoughts.

Reply:
What @sr525 is saying is that `plt.subplots` is *bad*. You want to instead use something like this https://github.com/lsst/analysis_tools/blob/eed07097883c01ff4d95e4b172ae7cabdcadfd7d/python/lsst/analysis/tools/actions/plot/skyPlot.py#L219-L220

Reply:
Regarding the `plt.rc_context()` that is fine, but you should instead do `from matplotlib import rc_context`.  You should not ever do `import matplotlib.pyplot as plt` in any pipeline code if at all possible.

Reply:
Ah, got it! I was confused because `rcparams` came up on Jira.

Reply:
I made those changes to see if it helps speed things up. The `make_fig` approach caused an error downstream:
 
 ```
  File "/home/en7908/repos/analysis_tools/python/lsst/analysis/tools/actions/plot/propertyMapPlot.py", line 827, in makePlot
    cbar = sp.draw_colorbar(**{"location": "top", "pad": 0.2, **plotConfig.colorbarKwargs})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/gpfs/HSC/LSST/stack_20240827/conda/envs/lsst-scipipe-9.0.0/lib/python3.11/site-packages/skyproj/_skyproj.py", line 1486, in draw_colorbar
    cbar = plt.colorbar(ax=ax, location=location, ticks=ticks,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 RuntimeError: No mappable was found to use for colorbar creation. First define a mappable such as an image (with imshow) or a contour set (with contourf).
 ```
 Looks like it doesn't like the axis created by `ax = fig.add_subplot(111)`.

Reply:
This should never have worked, it was just a coincidence.  This should be `ax=sp.ax` to get the correct axis.

Reply:
Also you do not want to use `plt.colorbar()`.  Hold on ...

Reply:
You can use `fig.colorbar()`.  Don't use `plt` at all!  Anywhere you have that it is going to be a problem.

Reply:
The `plt.colorbar()` line is from `_skyproj.py`.

Reply:
Ha!  Okay, well I have that fixed on skyproj 2 (beta), which I'll be testing with the stack after commissioning is over.  Sorry about that!  But this error is now coming from within skyproj?  Ugh, hold on...

Reply:
Thank you for the help Eli.